rhine-terminal/TerminalSimple.hs

```diff
@@ -0,0 +1,100 @@
+{- | Example application for the @rhine-terminal@ library. -}
+
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+import Prelude hiding (putChar)
+
+-- rhine
+import Control.Monad.Schedule ()
+
+import Data.Text (Text)
+import qualified Data.Text as T
+import System.Terminal
+import System.Terminal.Internal
+import FRP.Rhine
+import System.IO hiding (putChar)
+
+import FRP.Rhine.Terminal (TerminalEventClock(..))
+import System.Exit (exitSuccess)
+
+-- Clocks
+
+data Input = Char Char
+           | Space | Backspace | Enter
+           | Exit
+
+type InputClock = SelectClock (TerminalEventClock LocalTerminal) Input
```

Comment:
In principle this clock could go in the library as well.

Reply:
I'm not so sure about this. The value returned by `awaitEvent` is [of type `Either Interrupt Event`](https://hackage.haskell.org/package/terminal-0.2.0.0/docs/System-Terminal.html#g:13) and I think it should be up to the user to decide to filter this source and which data types to map to. In this example I map onto a kitchen sink `Input` ADT  (which also maps any interrupt to `Exit`, while in the more [involved example](https://github.com/turion/rhine/pull/175/files#diff-86197bb5ec43f3df9e0f77886bf50cbc3cbc651bb977058389cd29e970b2894dR46) interrupts have their own `SignalClock`.

Reply:
Ok, good point. Then this serves as a good example how users can customize the type returned by a clock.